### PR TITLE
test(artifacts test): initial support for Amazon Linux 2

### PR DESCRIPTION
**Pull request by enaydanov**
Trello: https://trello.com/c/BoouY9P3/1549-support-for-amazon-linux-2

Key things in the puzzle:
- root block device should be `/dev/xvda`
- add `--no-cpuscaling-setup` to `scylla_setup`

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I gave variables/functions meaningful self-explanatory names
- [x] I didn't leave commented-out/debugging code
- [x] I didn't copy-paste code
- [x] I added the relevant `backport` labels
- [x] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [x] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [x] All new and existing unit tests passed (CI)
- [x] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/cluster.py [hunk 1]
```diff
@@ -2040,6 +2040,7 @@ def scylla_setup(self, disks):
         """
         result = self.remoter.run('/sbin/ip -o link show |grep ether |awk -F": " \'{print $2}\'', verbose=True)
         devname = result.stdout.strip()
+        extra = self.parent_cluster.params.get('append_scylla_setup_args')
```
Comment (fruch) [inline sdcm/cluster.py]: `extra` is a bit vague.

maybe `extra_setup_args` would be better ?

**Comment by fruch** [comment]
we are going to be building the AMIs based on this Amazon Linux, 
won't it be enough to test the AMIs we produce ?

**Comment by enaydanov** [comment]
> we are going to be building the AMIs based on this Amazon Linux,
> won't it be enough to test the AMIs we produce ?

I hope so. But I'm not sure about IPv6 things you guys told.

We don't need to care about `--no-cpuscaling-setup` option because it should be resolved by guys who responsible for `scylla_setup` script, but we should use correct root block device in our tests. Right now, we'll fail if we start a AMI job against Scylla AMI based on Amazon Linux because of wrong block device.

We should be smarter here and change behavior to automatically get root device name from the AMI used:

```python
>>> boto3.resource("ec2", "eu-west-1").Image("ami-099a8245f5daa82bf").root_device_name
'/dev/xvda'
``` 

This thing looks like separate task and shouldn't be resolved here.

File: sdcm/cluster.py (hunk 1 above)
Comment (enaydanov) [reply sdcm/cluster.py]: changes

File: test-cases/artifacts/amazon2.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+ami_db_scylla_user: 'ec2-user'
+ami_id_db_scylla: 'ami-099a8245f5daa82bf'
+append_scylla_setup_args: '--no-cpuscaling-setup'
```
Comment (bentsi) [inline test-cases/artifacts/amazon2.yaml]: is this official for Amazon Linux?

Comment (fruch) [reply test-cases/artifacts/amazon2.yaml]: there's no yet official instructions for installation on  Amazon Linux

Comment (fruch) [reply test-cases/artifacts/amazon2.yaml]: in https://github.com/scylladb/scylla/pull/5540, seem like @syuu1228 doesn't use this option, and can install scylla just fine... (maybe other option for the AMI also disable the same thing)

Comment (bentsi) [reply test-cases/artifacts/amazon2.yaml]: is this temporary, until it will be fixed?

Comment (bentsi) [reply test-cases/artifacts/amazon2.yaml]: If this is an issue need to open one in relevant repo

Comment (amoskong) [inline test-cases/artifacts/amazon2.yaml]: @enaydanov what's the error in cpuscaling setup? do we have scylla issue for that?

File: test-cases/artifacts/amazon2.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+n_db_nodes: 1
+n_loaders: 0
+n_monitor_nodes: 0
+nemesis_class_name: 'NoOpMonkey'
+region_name: 'eu-west-1'
+scylla_linux_distro: 'centos'
+scylla_repo: 'https://s3.amazonaws.com/downloads.scylladb.com/rpm/centos/scylla-3.3.repo'
+store_results_in_elasticsearch: False
+system_auth_rf: 1
+test_duration: 600
```
Comment (amoskong) [inline test-cases/artifacts/amazon2.yaml]: The artifact-test won't take too much time, 600 mins is too big.
The artifact-tests of master took about 18 mins: https://jenkins.scylladb.com/job/scylla-master/job/artifacts/

30 mins should be enough for artifact-test test_duration.

Comment (enaydanov) [reply test-cases/artifacts/amazon2.yaml]: We had an issue with OEL job: #1889 . 30 minutes is not enough sometimes. But I agree that 10 hours is redundant for artifacts test. I'll change it to 60.

File: test-cases/artifacts/amazon2.yaml (hunk 2 above)
Comment (enaydanov) [reply test-cases/artifacts/amazon2.yaml]: Filed https://github.com/scylladb/scylla/issues/5977

Comment (enaydanov) [reply test-cases/artifacts/amazon2.yaml]: There is no `cpupower.service` file available on Amazon Linux 2. Filed https://github.com/scylladb/scylla/issues/5977

**Pull request merged**